notebooks/working/6.0-mab-estimate-causal-effect.md

```diff
@@ -2166,32 +2178,166 @@ for sim_size, number in zip(simulation_sizes, sim_number):
+    long_sim_data_causal.loc[long_sim_data_causal['mode_id']==4,'total_travel_cost'] = \
+                                                 (WTW_Reg['total_travel_cost_on_total_travel_time'].params[0] +\
+                                                  WTW_Reg['total_travel_cost_on_total_travel_time'].params[1] *\
+                                                  long_sim_data_causal.loc[long_sim_data_causal['mode_id']==4,'total_travel_time'] +\
+                                                  np.random.normal(loc=0,
+                                                                   scale = WTW_Reg['total_travel_cost_on_total_travel_time'].resid.std(),
+                                                                   size = long_sim_data_causal.loc[long_sim_data_causal['mode_id']==4].shape[0]))
+    
+    
+    ############ Drive-Transit-Walk ############
+    long_sim_data_causal.loc[long_sim_data_causal['mode_id']==5,'total_travel_cost'] = \
+                                                 (DTW_Reg['total_travel_cost_on_total_travel_time'].params[0] +\
+                                                  DTW_Reg['total_travel_cost_on_total_travel_time'].params[1] *\
+                                                  long_sim_data_causal.loc[long_sim_data_causal['mode_id']==5,'total_travel_time'] +\
+                                                  np.random.normal(loc=0,
+                                                                   scale = DTW_Reg['total_travel_cost_on_total_travel_time'].resid.std(),
+                                                                   size = long_sim_data_causal.loc[long_sim_data_causal['mode_id']==5].shape[0]))
+    
+    ############ Walk-Transit-Drive ############
+    long_sim_data_causal.loc[long_sim_data_causal['mode_id']==6,'total_travel_cost'] = \
+                                                 (WTD_Reg['total_travel_cost_on_total_travel_time'].params[0] +\
+                                                  WTD_Reg['total_travel_cost_on_total_travel_time'].params[1] *\
+                                                  long_sim_data_causal.loc[long_sim_data_causal['mode_id']==6,'total_travel_time'] +\
+                                                  np.random.normal(loc=0,
+                                                                   scale = WTD_Reg['total_travel_cost_on_total_travel_time'].resid.std(),
+                                                                   size = long_sim_data_causal.loc[long_sim_data_causal['mode_id']==6].shape[0]))
+    
+    
+    ############ Compute Estimated Probabilities ############
+    estimated_probabilities = mnl_model_sim.predict(long_sim_data_causal)
+    long_sim_data_causal['estimated_probabilities'] = estimated_probabilities
+    
+    ############ True Probabilities ############
+    true_probabilities = mnl_model.predict(long_sim_data_causal)
+    long_sim_data_causal['true_probabilities'] = true_probabilities
```

Comment:
@hassanobeid1994 and @bouzaghrane:
I just want to double check that this is indeed the correct computation of the true probabilities, according to the model used to generate the data (as opposed to using the model estimated from the generated data).

It looks like it is, but I remember @hassanobeid1994 that you pointed out in one of our sunday calls that there was an issue where the wrong probabilities were being used. As far as I know, the issue was never recorded in an official comment on github, so we don't have a way of tracking if the issue was correctly and finally resolved.

This is my attempt at creating such a paper trail. @hassanobeid1994, can you:
- review this pull-request one more time,
- respond to this comment to confirm that everything is correct, and
- formally approve this pull-request?